pkg/virt-controller/watch/vmi.go

```diff
@@ -200,6 +207,62 @@ func (e *syncErrorImpl) Reason() string {
 	return e.reason
 }
 
+type cidsMap struct {
```

Comment:
how about moving the cid allocator in its own files `vsock.go` and add some comprehensive unit-tests in `vsock_test.go`? This one should really work and should not be broken by accident.

Reply:
Done